Resources/public/js/views/dashboard/ez-dashboardblockallcontentview.js

```diff
@@ -0,0 +1,157 @@
+
+    var BLOCK_IDENTIFIER = 'all-content',
+        CLASS_LOADING = 'is-loading',
+        CLASS_ROW_SELECTED = 'is-row-selected',
+        SELECTOR_ROW = '.ez-allcontent-block-row',
+        SELECTOR_CONTENT = '.ez-allcontent-block-content',
+        EVENTS = {};
+
+    EVENTS[SELECTOR_ROW] = {tap: '_uiShowRowOptions'};
+
+    /**
+     * The all content dashboard block view
+     *
+     * @namespace eZ
+     * @class DashboardBlockAllContentView
+     * @constructor
+     * @extends eZ.DashboardBlockBaseView
+     */
+    Y.eZ.DashboardBlockAllContentView = Y.Base.create('dashboardBlockAllContentView', Y.eZ.DashboardBlockBaseView, [Y.eZ.AsynchronousView], {
+        initializer: function () {
+            this._fireMethod = this._getContentData;
+            /**
+             * Stores the click outside event handler
+             *
+             * @property _clickOutsideHandler
+             * @type {Object}
+             * @protected
+             */
+            this._clickOutsideHandler = null;
+            this.events = Y.merge(this.events, EVENTS);
+
+            this._set('identifier', BLOCK_IDENTIFIER);
+            this.get('container').addClass(this._generateViewClassName(Y.eZ.DashboardBlockBaseView.NAME));
+        },
+
+        /**
+         * Renders the dashboard block view
+         *
+         * @method render
+         * @return {eZ.DashboardBlockAllContentView} the view itself
+         */
+        render: function () {
+            this.get('container').setHTML(this.template()).addClass(CLASS_LOADING);
+
+            return this;
+        },
+
+        /**
+         * Shows the row options
+         *
+         * @method _uiShowOptions
+         * @protected
+         * @param event {Object} event facade
+         */
+        _uiShowRowOptions: function (event) {
+            this._uiHideRowOptions();
+
+            event.currentTarget.addClass(CLASS_ROW_SELECTED);
+
+            this._clickOutsideHandler = this.get('container').one(SELECTOR_CONTENT).on('clickoutside', Y.bind(this._uiHideRowOptions, this));
+        },
+
+        /**
+         * Clears the click outside handler
+         *
+         * @method _clearClickOutsideHandler
+         * @protected
+         */
+        _clearClickOutsideHandler: function () {
+            if (this._clickOutsideHandler) {
+                this._clickOutsideHandler.detach();
+            }
+        },
+
+        /**
+         * Hides the row options
+         *
+         * @method _uiHideOptions
+         * @protected
+         */
+        _uiHideRowOptions: function () {
+            this.get('container').all(SELECTOR_ROW).removeClass(CLASS_ROW_SELECTED);
+
+            this._clearClickOutsideHandler();
+
+        },
+
+        /**
+         * Makes request for data
+         *
+         * @method _getContentData
+         * @protected
+         * @param event {Object} event facade
+         */
+        _getContentData: function () {
+            this.fire('contentSearch', {
```

Comment:
> I'm still not convinced. The root location is always the top node of content tree.

There are also content items under media and users. We have also mentioned, together with André, that it would technically be possible in some implementations to have _more_, custom top level trees. 

How come this hasn't been solved ? I think that Damien has provided plenty of arguments, and explicitly _asked_ that it is done differently. Unless there is a good _technical_ reason not to, it should be done.